### PR TITLE
Refactor Server encapsulating board functionality

### PR DESCRIPTION
**Pull request by lmarquezMS**
@andresmoschini , @mravinale.

Was separated the board's responsibilities on server creating a Board Service module.
This module works acting on differents events produced by server.
Also was created a few unit tests to verify the following functionalities:
- On a new connection with the server, it returns the board.
- On a change on the board sended to server, the change is saved on Server board.

### CODE REVIEW

**Comment by andresmoschini** [comment]
@lmarquezMS do you think that it is possible to rebase it removing WIP commits and generating meaningful ones? let me know if you need help on that.

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@ coverage/
 node_modules/
 typings/
 npm-debug.log
+prueba/
```
Comment (andresmoschini) [inline .gitignore:5]: Is it necessary?

File: client/koclient.html [hunk 2]
```diff
@@ -21,7 +21,7 @@
             <a href="#" class="close"
             data-bind="click: $parent.removeNote">x</a>
             <h2  data-bind="jeditable: title, jeditableOptions: {}">Title</h2>
-            <div class="text-editable" data-bind="jeditable: content, jeditableOptions: {type: 'textarea', submit: 'Ok', cancel: 'Cancel'}" >Inner text</div>
+            <div class="text-editable" data-bind="jeditable: content, jeditableOptions: {type: 'textarea'}" >Inner text</div>
```
Comment (andresmoschini) [inline client/koclient.html:24]: How is it related with current PR?

File: gulpfile.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+    exec = require('child_process').exec,
+    gulp = require('gulp-help')(gulp),
+    tsconfig = require('gulp-tsconfig-files'),
+    path = require('path'),
+    inject = require('gulp-inject'),
+    gulpSequence = require('gulp-sequence'),
+    rename = require('gulp-rename'),
+    del = require('del'),
+    jasmine = require('gulp-jasmine'),
+    watch = require('gulp-watch');
```
Comment (andresmoschini) [inline gulpfile.js:12]: It is a matter of taste, but I am not sure how it is related to current PR scope.

File: .gitignore (hunk 1 above)
Comment (lmarquezMS) [inline .gitignore:5]: it's necessary to run the tests, this folder contains all compilated files.

Comment (andresmoschini) [inline .gitignore:5]: Could you change it by a nicer name?

File: gulpfile.js (hunk 3 above)
Comment (lmarquezMS) [inline gulpfile.js:12]: the change of the way that I rewrite the assignations it's not related with the PR. Despite that, jasmine and watch modules (line 11 and 12 respectively) are necessary to perform the tests.

File: prueba/common/client-app.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    rootNode: null,
+    socketEventName: "board",
+    throttlingInterval: 1000
+};
+function start(boardVM, config) {
+    var app = new App(boardVM, config);
+    app.start();
+    return app;
+}
+exports.start = start;
```
Comment (andresmoschini) [inline prueba/common/client-app.js]: If I am not wrong, you can avoid this line prepending original function declaration with `export`, for example:

```
export function start(boardVM, config) {
    var app = new App(boardVM, config);
```

Comment (andresmoschini) [inline prueba/common/client-app.js]: Sorry, it is a generated file. Ignore my previous message.

File: prueba/common/client-app.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+/// <reference path="../../node_modules/rfc6902/rfc6902.d.ts" />
+var rfc6902 = require("rfc6902");
```
Comment (andresmoschini) [inline prueba/common/client-app.js:2]: should these files (`prueba/**` ) be included in .gitignore?

**Comment by lmarquezMS** [comment]
> @lmarquezMS do you think that it is possible to rebase it removing WIP commits and generating meaningful ones? let me know if you need help on that.

I'll try to do that by myself, if this take me too long I'll tell you.

File: src/common/client-app.ts [hunk 6]
```diff
@@ -25,8 +25,7 @@ export function start(boardVM: IBoardVM, config?: AppConfiguration) {
   return app;
 }
 
-class App {
-  shadowServer: model.Board = { };
+export class App {
```
Comment (andresmoschini) [inline src/common/client-app.ts:28]: Why are there changes in client-app?

**Comment by lmarquezMS** [comment]
This pull request will be closed and redefined in isolated pr

**Pull request closed without merging**